src/gui/widgets/unit_preview_pane.cpp

```diff
@@ -98,6 +101,93 @@ static inline tree_view_node& add_name_tree_node(tree_view_node& header_node, co
 	return child_node;
 }
 
+template<typename TGet_resistance>
+static inline std::string get_hp_tooltip(const utils::string_map& res, const TGet_resistance& get)
+{
+	std::ostringstream tooltip;
+
+	std::set<std::string> resistances_table;
+
+	bool att_def_diff = false;
+	for (const utils::string_map::value_type &resist : res)
+	{
+		std::ostringstream line;
+		line << translation::gettext(resist.first.c_str()) << ": ";
+		// Some units have different resistances when attacking or defending.
+		int res_att = 100 - get(resist.first, true);
+		int res_def = 100 - get(resist.first, false);
+
+		if (res_att == res_def) {
+			line << "<span foreground=\"" << unit_helper::resistance_color(res_def) << "\">" << utils::signed_percent(res_def) << "</span>" << '\n';
```

Comment:
i just copied that code from https://github.com/wesnoth/wesnoth/blob/master/src/reports.cpp#L432